crates/uv-python/src/installation.rs

```diff
@@ -343,6 +345,57 @@ impl PythonInstallation {
     pub fn into_interpreter(self) -> Interpreter {
         self.interpreter
     }
+
+    pub(crate) fn warn_on_prerelease_python_upgrade_available(
+        &self,
+        request: &PythonRequest,
+        python_downloads_json_url: Option<&str>,
+    ) {
+        if request.allows_prereleases() {
+            return;
+        }
+
+        let interpreter = self.interpreter();
+        let version = interpreter.python_version();
+
+        if version.pre().is_none() {
+            return;
+        }
+
+        if !interpreter.is_managed() {
+            return;
+        }
+
+        if !interpreter
+            .implementation_name()
+            .eq_ignore_ascii_case("cpython")
+        {
+            return;
+        }
+
+        let release = version.only_release();
+
+        let Ok(download_request) = PythonDownloadRequest::try_from(&interpreter.key()) else {
+            return;
+        };
+
+        let download_request = download_request.with_prereleases(false);
+
+        if download_request.has_stable_download_at_least(&release, python_downloads_json_url) {
```

Comment:
I'm a little confused that we're not just doing `request.iter_downloads()` and looking for a stable download?

Reply:
Yeah I can inline this, `has_stable_download_at_least` is doing exactly that.